### PR TITLE
Add presubmit job and periodic job for system test of kube-api

### PR DESCRIPTION
**Pull request by osherdp**
Right now, it's not including nmstate mechanism.

### CODE REVIEW

File: ci-operator/config/openshift/assisted-service/openshift-assisted-service-master.yaml [hunk 1]
```diff
@@ -143,6 +143,24 @@ tests:
...
     workflow: baremetalds-assisted-install
+- as: e2e-metal-assisted-kube-api
+  steps:
+    cluster_profile: packet
+    dependencies:
+      ASSISTED_SERVICE_IMAGE: pipeline:assisted-service
+    env:
+      ASSISTED_CONFIG: |
+        ENABLE_KUBE_API=true
+        NUM_MASTERS=1
```
Comment (YuviGold) [inline ci-operator/config/openshift/assisted-service/openshift-assisted-service-master.yaml:154]: single node? Should be part of the name?

File: ci-operator/config/openshift/assisted-service/openshift-assisted-service-master.yaml [hunk 2]
```diff
@@ -143,6 +143,24 @@ tests:
...
+        NUM_WORKERS=0
+        IPv6=yes
+        IPv4=no
+        PROXY=yes
+        ISO_IMAGE_TYPE=minimal-iso
+        KUBE_API=yes
```
Comment (YuviGold) [inline ci-operator/config/openshift/assisted-service/openshift-assisted-service-master.yaml:160]: Why there are both `ENABLE_KUBE_API` and `KUBE_API`?

File: ci-operator/config/openshift/assisted-service/openshift-assisted-service-master.yaml (hunk 1 above)
Comment (osherdp) [reply ci-operator/config/openshift/assisted-service/openshift-assisted-service-master.yaml:154]: currently kube-api is targeted to single-node only
but the single-node is not part of the thing here, we're mainly interested in the kube-api interface here

File: ci-operator/config/openshift/assisted-service/openshift-assisted-service-master.yaml (hunk 2 above)
Comment (YuviGold) [reply ci-operator/config/openshift/assisted-service/openshift-assisted-service-master.yaml:160]: This naming is really unclear...

Comment (osherdp) [reply ci-operator/config/openshift/assisted-service/openshift-assisted-service-master.yaml:160]: Yes, we will need to discuss some renamings

**Pull request merged**